File: trueshot-backend/src/main/java/com/ongtangco/trueshot/model/Product.java
package com.ongtangco.trueshot.model;

import com.ongtangco.trueshot.enums.ConditionGrade;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Product {
    private Integer id;
    private String sku;
    private String brand;
    private String modelName;
    private ConditionGrade conditionGrade;
    private String description;
    private String imageUrl;
    private String lensMount;
    private String sensorType;
    private Integer releaseYear;
    private Integer shutterCount;
    private Integer stockQuantity;
    private BigDecimal price;
    private Boolean featured;
    private Boolean available;
    private LocalDateTime createdAt;
    private LocalDateTime lastUpdated;
}

File: trueshot-frontend/src/app/models/product.ts
export interface Product {
  id: number;
  sku: string;
  brand: string;
  modelName: string;
  conditionGrade: string;
  description: string;
  imageUrl: string;
  lensMount: string;
  sensorType: string;
  releaseYear: number | null;
  shutterCount: number | null;
  stockQuantity: number;
  price: number;
  featured: boolean;
  available: boolean;
  createdAt: string;
  lastUpdated: string;
}
